pkg/imgpkg/bundle/bundle_images_lock.go

```diff
@@ -67,24 +97,78 @@ func (o *Bundle) buildAllImagesLock(throttleReq *util.Throttle, processedImgs *p
 		}()
 	}
 
-	for range imagesLock.Images {
+	for range imageRefs {
 		if err := <-errChan; err != nil {
+			return nil, nil, err
+		}
+	}
+
+	return bundles, resultImagesLock, nil
+}
+
+func (o *Bundle) fetchImagesRef(img regv1.Image, resultImagesLock *ImagesLock, logger util.LoggerWithLevels) (ImagesRef, error) {
+	// Check locations
+	bundleDigestRef, err := regname.NewDigest(o.DigestRef())
+	if err != nil {
+		panic(fmt.Sprintf("Internal inconsistency: The Bundle Reference '%s' does not have a digest", o.DigestRef()))
+	}
+
+	var imageRefs ImagesRef
+	locationsConfig, err := NewLocations(logger).Fetch(o.imgRetriever, bundleDigestRef)
+	if err == nil {
+		imageRefs = o.processLocations(resultImagesLock, locationsConfig)
+	} else if _, ok := err.(*LocationsNotFound); !ok {
+		return nil, err
+	} else {
+		imagesLock, err := o.imagesLockReader.Read(img)
+		if err != nil {
 			return nil, err
 		}
+
+		currentImagesLock := NewImagesLock(imagesLock, o.imgRetriever, o.Repo())
+		imageRefs = currentImagesLock.ImagesRefs()
+	}
```

Comment:
i think we should always read ImagesLock (it's still source of truth for referenced images by the bundle), but if we also find locations then we should AddLocation to the image ref.

Reply:
Done